src/core/server/base.js

```diff
@@ -109,11 +122,11 @@ function baseServer(routes, createStore, { appInstanceName = appName } = {}) {
 
       if (err) {
         log.error({ err, req });
-        return res.status(500).end(errorString);
+        return showErrorPage(res, 500);
       }
 
       if (!renderProps) {
-        return res.status(404).end('Not found.');
+        return showErrorPage(res, 404);
```

Comment:
Yeah, a proper page would be great - there's another issue somewhere for that. Needs some UX input. I was hoping to be able to use fairly self-contained pages everywhere for 4xx / 5xx